presto-main/src/test/java/com/facebook/presto/sql/relational/TestRowExpressionTranslator.java

```diff
@@ -251,43 +269,47 @@ public TranslatedExpression<String> translateVariable(VariableReferenceExpressio
     public static class TestFunctions
     {
         @ScalarFunction
-        @SqlType(StandardTypes.BIGINT)
-        public static String bitwiseAnd(@SqlType(StandardTypes.BIGINT) String left, @SqlType(StandardTypes.BIGINT) String right)
+        @SupportedSignatures(@SqlSignature(argumentType = StandardTypes.BIGINT, returnType = StandardTypes.BIGINT))
+        public static String bitwiseAnd(String left, String right)
         {
             return left + " BITWISE_AND " + right;
         }
 
         @ScalarFunction("ln")
-        @SqlType(StandardTypes.DOUBLE)
-        public static String ln(@SqlType(StandardTypes.DOUBLE) String sql)
+        @SupportedSignatures(@SqlSignature(argumentType = StandardTypes.DOUBLE, returnType = StandardTypes.DOUBLE))
+        public static String ln(String sql)
         {
             return "LNof(" + sql + ")";
         }
 
         @ScalarFunction("ceil")
-        @SqlType(StandardTypes.DOUBLE)
-        public static String ceil(@SqlType(StandardTypes.BOOLEAN) String sql)
+        @SupportedSignatures(@SqlSignature(argumentType = StandardTypes.BOOLEAN, returnType = StandardTypes.DOUBLE))
+        public static String ceil(String sql)
         {
             return "CEILof(" + sql + ")";
         }
 
         @ScalarFunction("not")
-        @SqlType(StandardTypes.BOOLEAN)
-        public static String not(@SqlType(StandardTypes.BOOLEAN) String sql)
+        @SupportedSignatures(@SqlSignature(argumentType = StandardTypes.BOOLEAN, returnType = StandardTypes.BOOLEAN))
+        public static String not(String sql)
         {
             return "NOT_2 " + sql;
         }
 
         @ScalarOperator(OperatorType.ADD)
-        @SqlType(StandardTypes.BIGINT)
-        public static String plus(@SqlType(StandardTypes.BIGINT) String left, @SqlType(StandardTypes.BIGINT) String right)
+        @SupportedSignatures({
+                @SqlSignature(argumentType = StandardTypes.INTEGER, returnType = StandardTypes.INTEGER),
+                @SqlSignature(argumentType = StandardTypes.BIGINT, returnType = StandardTypes.BIGINT),
+                @SqlSignature(argumentType = "decimal(38, 0)", returnType = "decimal(38, 0)"),
+                @SqlSignature(argumentTypes = {StandardTypes.BIGINT, StandardTypes.INTEGER}, returnType = StandardTypes.BIGINT)})
+        public static String plus(String left, String right)
```

Comment:
How are you asserting these are applied correctly?

Reply:
testBasicOperator